kytos/core/buffers/buffers.py

```diff
@@ -55,6 +74,7 @@ async def aput(self, event):
             event (:class:`~kytos.core.events.KytosEvent`):
                 KytosEvent sent to queue.
         """
+        await self.check_put_rate_limits(event)
```

Comment:
Have you tested this with multiple rate limiters and different identifiers? The underlying `await asyncio.gather(*coros)` if any of the rate limiters end up limiting it'll end up awaiting for them all and since `aput` is called by NApp looks like multiple rate limiters will interfere with each other, from what I followed looks like this is what will happen here, I didn't run this locally, let me know if I'm seeing this incorrectly. 

One of the main use cases for rate limiting before putting into the queue as close as the call site as possible is to also not impact on other NApps, for this to work here without interference I believe it would need an approach that you mentioned sometime to either add "meta" information to derive a producer namespace/context (but maybe it's too magic). Let me know if you can see any solution without too much refactoring for this.

Reply:
Maybe we can also consider nuking producer rate limiters configuration in the core and only maintain a registration of rate limiter producers, and then NApps would register/create a rate limit data class instance `EventRateLimiter` and then in the call site instead of calling `aput/put` they'd call `aput_rl(event, **kwargs)` or `put_rl(event, **kwargs)` - or any other better named methods - which would make use of the registered `EventRateLimiter` (assuming that'd be have a blocking version functioning too), and then they could create this on the fly - during each NApp `setup` - and use the rate limit in the context that the producer needs to be limited (making it free to also be reshared in the same NApp or whatever boundary it makes sense for the call sites that need to rate limit that type of event). Wdyt? 

Let me know if you can see something simpler that solves this problem without fully committing to a producer config on kytos.conf yet, maybe that's also not mutually exclusive but for the NApps, I believe the most frequent use case that'd be the one each NApp regsitering then using the rate limiter.